assets/js/blocks-registry/payment-methods/extensions-config.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import { CanMakePayment } from '@woocommerce/type-defs/payments';
+
+type CanMakePaymentExtensionCallbacks = Record< string, CanMakePayment >;
+export type ExtensionNamespace = keyof CanMakePaymentExtensionsCallbacks;
+export type PaymentMethodName = keyof CanMakePaymentExtensionCallbacks;
+
+export type CanMakePaymentExtensionsCallbacks = Record<
+	PaymentMethodName,
+	CanMakePaymentExtensionCallbacks
```

Comment:
The naming here is quite confusing. I'd suggest renaming `CanMakePaymentExtensionsCallbacks` to `NamespacedCanMakePaymentExtensionsCallbacks`

I also think the key is wrong, shouldn't it be?
```suggestion
export type CanMakePaymentExtensionsCallbacks = Record<
	ExtensionNamespace,
	CanMakePaymentExtensionCallbacks
```

Reply:
Great catch, thanks for spotting this, fixed in 930c6886baecbbb8ae388983197fcc9509053b9e